pkg/runtime/reconciler.go

```diff
@@ -359,6 +363,55 @@ func (r *reconciler) handleReconcileError(err error) (ctrlrt.Result, error) {
 	return ctrlrt.Result{}, err
 }
 
+// updateCRTerminalConditions updates status of CR for terminal conditions
```

Comment:
I don't believe this should be in the reconciler. The resourceManager should modify the `*resource` that is returned from one of its methods (like Update or ReadOne, etc) with a Condition indicating a terminal state... the reconciler's `updateCRStatus()` call, which should patch the CR's Status struct, should handle the call to k8sclient.Patch.